Collections/FISAppDelegate.m

```diff
@@ -10,40 +10,193 @@
 
 @implementation FISAppDelegate
 
-- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
-{
-    self.window = [[UIWindow alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
     // Override point for customization after application launch.
-    self.window.backgroundColor = [UIColor whiteColor];
-    [self.window makeKeyAndVisible];
+
     return YES;
 }
 
-- (void)applicationWillResignActive:(UIApplication *)application
-{
-    // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
-    // Use this method to pause ongoing tasks, disable timers, and throttle down OpenGL ES frame rates. Games should use this method to pause the game.
+/**
+ 
+ * Define your methods here.
+ 
+ */
+
+-(NSArray *)arrayByAscendingSortingArray:(NSArray *)array {
+    NSSortDescriptor *sortByNilAsc = [NSSortDescriptor sortDescriptorWithKey:nil ascending:YES];
+    
+    return [array sortedArrayUsingDescriptors:@[sortByNilAsc]];
+}
+
+-(NSArray *)arrayByDescendingSortingArray:(NSArray *)array {
+    NSSortDescriptor *sortByNilDesc = [NSSortDescriptor sortDescriptorWithKey:nil ascending:NO];
+    
+    return [array sortedArrayUsingDescriptors:@[sortByNilDesc]];
+}
+
+-(NSArray *)arrayBySwappingFirstObjectWithLastObjectInArray:(NSArray *)array {
+    NSMutableArray *mutableArray = [array mutableCopy];
+    NSUInteger endIndex = mutableArray.count - 1;
+    
+    [mutableArray exchangeObjectAtIndex:0 withObjectAtIndex:endIndex];
+    return [NSArray arrayWithArray:mutableArray];
+}
+
+-(NSArray *)arrayByReversingArray:(NSArray *)array {
+    NSMutableArray *mutableArray = [array mutableCopy];
+    NSUInteger i = 0;
+    NSUInteger j = mutableArray.count - 1;
+    
+    for ( ; i < (mutableArray.count / 2); i++) {
```

Comment:
I would put the i in the looping stuff. The j can be separte but that i=0 wants to be in the for declaration